cql3/expr/prepare_expr.cc

```diff
@@ -810,11 +810,29 @@ cast_prepare_expression(const cast& c, data_dictionary::database db, const sstri
     // Using the original receiver wouldn't work in such cases - it would complain
     // that untyped_constant(1234) isn't a valid blob constant.
     return cast{
+        .style = cast::cast_style::c,
         .arg = prepare_expression(c.arg, db, keyspace, schema_opt, cast_type_receiver),
         .type = receiver->type,
     };
 }
 
+static
+std::optional<expression>
+sql_cast_prepare_expression(const cast& c, data_dictionary::database db, const sstring& keyspace, const schema* schema_opt, lw_shared_ptr<column_specification> receiver) {
```

Comment:
We should eventually document the difference between what you called here "C" and "SQL" type casts (in a document and/or a comment). Or do we have such documentation already? Are these names you invented now or we used them somewhere before?

In `test/cql-pytest/test_cast*.py` I called these casts "query" and "data" casts respectively - what I called "query" casts can only change, losslessly, constants appearing in the query, whereas the "data" casts can do lossy conversions of data read from the database.

Reply:
The user-facing documentation calls the SQL-style cast "casting" and doesn't mention the C-style cast.

I don't propose we use these terms in user-facing documentation.